systems/subscriber/registry/pkg/db/subscriber_repo_test.go

```diff
@@ -56,60 +57,399 @@ func (u UkamaDbMock) ExecuteInTransaction2(dbOperation func(tx *gorm.DB) *gorm.D
 	return nil
 }
 
-func TestSubscriber_Add(t *testing.T) {
+func setupTestDB(t *testing.T) (*sql.DB, sqlmock.Sqlmock, int_db.SubscriberRepo) {
 	db, mock, err := sqlmock.New()
 	if err != nil {
 		t.Fatalf("an error '%s' was not expected when opening a stub database connection", err)
 	}
-	defer db.Close()
+
 	gdb, _ := gorm.Open(postgres.New(postgres.Config{
 		DSN:                  "sqlmock_db_0",
 		DriverName:           "postgres",
 		Conn:                 db,
 		PreferSimpleProtocol: true,
 	}), &gorm.Config{})
+
 	repo := int_db.NewSubscriberRepo(&UkamaDbMock{
 		GormDb: gdb,
 	})
-	dateStr := "07-03-2023"
 
-	subscriber := int_db.Subscriber{
+	return db, mock, repo
+}
+
+func createTestSubscriber(name, email string) int_db.Subscriber {
+	return int_db.Subscriber{
 		SubscriberId:          uuid.NewV4(),
-		Name:                  "John",
+		Name:                  name,
 		NetworkId:             uuid.NewV4(),
-		
+		Email:                 email,
 		PhoneNumber:           "555-555-5555",
 		Gender:                "Male",
-		DOB:                   dateStr,
+		DOB:                   "07-03-2023",
 		ProofOfIdentification: "Driver's License",
 		IdSerial:              "ABC123",
 		Address:               "123 Main St.",
 		CreatedAt:             time.Now(),
 		UpdatedAt:             time.Now(),
 		DeletedAt:             nil,
 	}
+}
+
+func TestSubscriber_Add(t *testing.T) {
+	t.Run("SuccessWithoutNestedFunc", func(t *testing.T) {
+		// Arrange
+		db, mock, repo := setupTestDB(t)
+		defer db.Close()
```

Comment:
We are not using the sqlmock.Db  so it is useless to have it returned, and therefore we should not have to defer it closure.

Reply:
addressed